test/scripts/hexo/validate_config.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const { spy } = require('sinon');
+
+describe('Validate config', () => {
+  const Hexo = require('../../../lib/hexo');
+  const hexo = new Hexo();
+  const validateConfig = require('../../../lib/hexo/validate_config');
+  const defaultConfig = require('../../../lib/hexo/default_config');
+  let logSpy;
+
+  beforeEach(() => {
+    logSpy = spy();
+    hexo.config = JSON.parse(JSON.stringify(defaultConfig));
+    hexo.log.warn = logSpy;
+    hexo.log.info = spy();
+  });
+
+  it('config.url - undefined', () => {
+    delete hexo.config.url;
+
+    try {
+      validateConfig(hexo);
+      should.fail();
+    } catch (e) {
+      e.name.should.eql('TypeError');
+      e.message.should.eql('Invalid config detected: "url" should be string!');
```

Comment:
`undefined` is practically an _empty_ value. If a user accidentally removes the value of `url:` or even the key itself, it's clearer to say `"url" should not be empty`.

Reply:
Although for YAML, `undefined` is usually caused by empty value. But Hexo also supports config in `json` format (Although most people just won't using json).

Anyway, I will update the error message then.